Hotfix 4.4.8

timbussmann opened the pull request:
see issue #2885

timbussmann commented:
@andreasohlund @weralabaj have a look at the commits

weralabaj commented on src/NServiceBus.sln.DotSettings:
```diff
@@ -122,7 +122,8 @@
 	<s:Boolean x:Key="/Default/CodeStyle/CodeFormatting/JavaScriptCodeFormatting/ALIGN_MULTIPLE_DECLARATION/@EntryValue">True</s:Boolean>
 	<s:Boolean x:Key="/Default/CodeStyle/CodeFormatting/JavaScriptCodeFormatting/JavaScriptFormatOther/ALIGN_MULTIPLE_DECLARATION/@EntryValue">True</s:Boolean>
-	<s:String x:Key="/Default/CodeStyle/CSharpFileLayoutPatterns/Pattern/@EntryValue">&lt;Patterns xmlns="urn:schemas-jetbrains-com:member-reordering-patterns"&gt;&#xD;
+	<s:String x:Key="/Default/CodeStyle/CSharpFileLayoutPatterns/Pattern/@EntryValue">&lt;?xml version="1.0" encoding="utf-16"?&gt;&#xD;
```
Not sure if we really want to make changes in this file now?

timbussmann replied:
nope, didn't want to commit that sneaky bastard

timbussmann replied:
changes come from the syncomatic update now

weralabaj commented on src/ObjectBuilder.Ninject/ObjectBuilder.Ninject.csproj:
```diff
@@ -101,13 +101,11 @@
-    
     <Error Condition="!Exists('..\packages\GitVersionTask.2.0.1\Build\GitVersionTask.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\GitVersionTask.2.0.1\Build\GitVersionTask.targets'))" />
     <Error Condition="!Exists('..\packages\NuGetPackager.0.5.4\build\NuGetPackager.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\NuGetPackager.0.5.4\build\NuGetPackager.targets'))" />
+    <Error Condition="!Exists('..\packages\Fody.1.29.3\build\portable-net+sl+win+wpa+wp\Fody.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Fody.1.29.3\build\portable-net+sl+win+wpa+wp\Fody.targets'))" />
```
As above - intentional?

timbussmann replied:
that was required with the fody update

weralabaj commented on src/NServiceBus.Core/Transports/Msmq/MsmqMessageSender.cs:
```diff
@@ -50,7 +51,9 @@ public void Send(TransportMessage message, Address address)
                                                                                 address.ToString()));
 
-                        if (UnitOfWork.HasActiveTransaction())
+                        bool suppressNativeTransactions;
```
Question - is there a situation when we want to allow for native transactions other than when dtc is enabled? If not then could we check for dtc enabled condition instead of using pipeline?
Or it won't be exactly the same effect?

andreasohlund replied:
>is there a situation when we want to allow for native transactions other than when dtc is enabled?

Native transactions is only in use when you're not using the DTC, can you clarify the question?

weralabaj replied:
I shouldn't write my questions in a hurry. @andreasohlund what I meant is, is there any scenario for removing timeouts where we want/need to rely on native transactions?
I was thinking about eliminating the pipeline flag (because that restricts us to 4.2+), but the only way I see we could do it is to change ISendMessagesInterface. Which is probably even a bigger mess.

andreasohlund commented on src/NServiceBus.AcceptanceTests/Timeouts/When_endpoint_uses_outdated_persistence_but_acknowledges.cs:
```diff
@@ -0,0 +1,40 @@
+            {
+                EndpointSetup<DefaultServer>(config =>
+                {
+                    config.SuppressOutdatedTimeoutPersistenceWarning();
```
Do we need a fullblown code api for this, since this will be a "dropin" replacement should we consider just a plan appsetting instead? `NServiceBus/Xyz`

timbussmann replied:
I'm not sure we can acceptance test it properly with only supporting config files and I also think we should offer as many options as possible to overcome that startup exception.

andreasohlund replied:
Just thinking to not force them to do code changes since this will likely hit old projects where they have "lost" the source code? ;)

timbussmann replied:
the check also allows for appsettings configuration so no code changes will be required.

andreasohlund replied:
Ok, fair enough

andreasohlund commented on src/NServiceBus.AcceptanceTests/Timeouts/When_endpoint_uses_outdated_timeout_persistence_with_dtc.cs:
```diff
@@ -0,0 +1,38 @@
+                EndpointSetup<DefaultServer>(config =>
+                {
+                    config.Configurer.ConfigureComponent<OutdatedTimeoutPersister>(DependencyLifecycle.SingleInstance);
+                    Configure.Transactions.Advanced(s => s.EnableDistributedTransactions());
```
Note: Pretty sure that this is the default but I think the tests becomes more clear when its there

timbussmann replied:
that was exactly the intention :)

andreasohlund commented on src/NServiceBus.Core/Persistence/Raven/TimeoutPersister/RavenTimeoutPersistence.cs:
```diff
@@ -10,7 +10,7 @@ namespace NServiceBus.Persistence.Raven.TimeoutPersister
     using Timeout.Core;
 
-    public class RavenTimeoutPersistence : IPersistTimeouts
+    public class RavenTimeoutPersistence : IPersistTimeouts, IPersistTimeoutsV2
```
Didn't realize RavenDB was still in the core in this version. So in essence we will properly support RavenDB timeouts in v4.4

timbussmann replied:
looks so, although the NSB.RavenDB package supports back to Core 4.1. Should we change that?

andreasohlund commented on src/NServiceBus.Core/Timeout/Core/TimeoutPersistenceVersionCheck.cs:
```diff
@@ -0,0 +1,78 @@
+        }
+    }
+
+    public class TimeoutPersistenceVersionCheck : IWantToRunWhenConfigurationIsComplete
```
Make this internal?

andreasohlund commented on src/ObjectBuilder.Unity/packages.config:
```diff
@@ -1,7 +1,7 @@
 <packages>
   <package id="CommonServiceLocator" version="1.0" targetFramework="net40" />
-  <package id="Fody" version="1.25.0" targetFramework="net40" developmentDependency="true" />
+  <package id="Fody" version="1.29.3" targetFramework="net40" developmentDependency="true" />
```
Why did you update Fody?

SimonCropp replied:
VS 2015 = roslyn = breaking change in the IL metadata spec = updated fody

andreasohlund replied:
Ok, got it!

timbussmann commented:
ok, tests work again now. Not going to fix the code inspection errors. Recent changes:
* fixed ScheduledTask naming because of Roslyn
* prefixed config file "suppression" setting with "NServiceBus/..."
* updated via settings syncomatic because of the build agents

@Particular/nservicebus please review

timbussmann commented:
I was thinking about adding a doco page about the startup check and how to overcome it which we could link from the exception message. The exception message is already quite long and not trivial to understand what's the issue, who's affected and how to disable the warning. Thoughts?

weralabaj commented:
@timbussmann @andreasohlund @SimonCropp 
Even though that's a potential message loss I don't see any similar content in the docs, do you think we should go beyond announcement and good release notes in this case?
However, this http://docs.particular.net/nservicebus/persistence/authoring-custom#timeout-persister definitely needs an update and explanation might get long. Today I'll create draft for this page and for announcement.
